src/components/Icon/index.tsx

```diff
@@ -1,73 +1,81 @@
-import PropTypes from 'prop-types';
 import React, {PureComponent} from 'react';
-import {View} from 'react-native';
+import {View, ViewStyle} from 'react-native';
 import styles from '@styles/styles';
 import * as StyleUtils from '@styles/StyleUtils';
 import themeColors from '@styles/themes/default';
 import variables from '@styles/variables';
 import IconWrapperStyles from './IconWrapperStyles';
 
-const propTypes = {
+type SrcProps = {
+    width?: number;
+    height?: number;
+    fill?: string;
+    hovered?: string;
+    pressed?: string;
+};
+
+type IconProps = {
     /** The asset to render. */
-    src: PropTypes.func.isRequired,
+    src: (props: SrcProps) => React.ReactNode;
 
     /** The width of the icon. */
-    width: PropTypes.number,
+    width?: number;
 
     /** The height of the icon. */
-    height: PropTypes.number,
+    height?: number;
 
     /** The fill color for the icon. Can be hex, rgb, rgba, or valid react-native named color such as 'red' or 'blue'. */
-    fill: PropTypes.string,
+    fill?: string;
 
     /** Is small icon */
-    small: PropTypes.bool,
+    small?: boolean;
 
     /** Is inline icon */
-    inline: PropTypes.bool,
+    inline?: boolean;
 
     /** Is icon hovered */
-    hovered: PropTypes.bool,
+    hovered?: boolean;
 
     /** Is icon pressed */
-    pressed: PropTypes.bool,
+    pressed?: boolean;
 
     // eslint-disable-next-line react/forbid-prop-types
-    additionalStyles: PropTypes.arrayOf(PropTypes.object),
-};
-
-const defaultProps = {
-    width: variables.iconSizeNormal,
-    height: variables.iconSizeNormal,
-    fill: themeColors.icon,
-    small: false,
-    inline: false,
-    additionalStyles: [],
-    hovered: false,
-    pressed: false,
+    additionalStyles?: ViewStyle[];
```

Comment:
Can we use here` StyleProp<ViewStyle>` ?

Reply:
It's always an array, so I guess not this time, hence the name `additionalStyle**s**`

Reply:
According to our guide, we should always use it without an array wrapper - https://github.com/Expensify/App/blob/main/contributingGuides/TS_CHEATSHEET.md#style-props

`StyleProp` already considers this

Reply:
Updated!